com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/delegates/dependency/CopybookDependencyServiceImpl.java

```diff
@@ -160,4 +228,71 @@ private Path createDependencyFileFolder() {
       return folderPath;
     }
   }
+
+  /**
+   * This method is used to search for a copybook against a given configuration of datasets that
+   * represent the sub-path of the copyooks folder
+   *
+   * @param filename copybook name
+   * @return The path of the existent copybook or null if not found
+   */
+  private Path findCopybook(String filename, String profile, List<String> datasetList) {
+    return retrievePathOrNull(filename, generatePathListFromSettings(profile, datasetList));
+  }
+
+  private Path retrievePathOrNull(String filename, List<Path> datasetPathList) {
+    return datasetPathList.stream()
+        .map(it -> applySearch(filename, it))
+        .filter(Objects::nonNull)
+        .findAny()
+        .orElse(null);
+  }
+
+  /**
+   * @param fileName copybook name
+   * @param targetFolderPath physical path of workspace where to search for the copybook
+   * @return Path of the found copybook in the target folder.
+   */
+  private Path applySearch(String fileName, Path targetFolderPath) {
+    try (Stream<Path> pathStream =
```

Comment:
We'll receive a code smell because of the following code is a duplication of `CopybookServiceImpl`

Reply:
Andrea's implementation which will merge my branch will cover this, that's why I didn't modify it